pages/article/ArticleHandler.inc.php

```diff
@@ -80,10 +81,10 @@ function initialize($request, $args = array()) {
 
 		import('classes.issue.IssueAction');
 		$issueAction = new IssueAction();
-		$context = $request->getContext();
+		$journal = $request->getJournal();
```

Comment:
Request::getContext is deprecated. Left it as it was on the stable-3_2_1 branch, shouldn't make any difference as `getContext` also returns an instance of a `Journal` (was confusing to me).

Reply:
The `getContext` function isn't deprecated (more specifically static calls to `Request::getContext` are, but instance calls `$request->getContext` are not). This is the generic name for the function to get the `Journal` in OJS, `Press` in OMP, etc., and it's pretty heavily used in `pkp-lib`. Personally, I favour using it over the more specific call, especially for anything that'll exist in multiple places (e.g. both OJS and OPS). However, they are equivalent.

Reply:
Understood. Does it make sense to specify return types of methods of the app-specific classes that have app non-specific parents? I mean the same that I did for `ArticleHandler::initialize()`?

Reply:
Broadly speaking, I'm in favour of adopting the conventions set in `pkp-lib` and reducing the amount of application-specific extension. Gradually we're trying to shrink the amount of code in each application repository and get rid of the subclasses where possible.

Reply:
Yeah I agree with Alec here. The context/journal distinction is confusing, but we're standardising on using the term "context" in all cases, and eventually we'll be able to align more of the class names and things.

For now, can you convert the changes you've made from `$journal` back to `$context` and `getJournal` back to `getContext`?